### ISSUE TITLE
FileSubscriber completes with succesful Promise if it can't open the file

### ISSUE THREAD

**Issue by oker1** [akka/akka#24889]
If you create a sink with ```FileIO.toPath("file.txt")``` and the file can't be opened the stream will complete, and a the Future[IOResult] will be successful with zero bytes written. This behavior can hide errors, in our broadcast flow everything seems normal if no bytes were written to the file and the other branches completed successfully.

**Comment by johanandren** [comment]
I think this was fixed in 2.5.10 (#23951). Which version are you seeing it in?

**Comment by oker1** [comment]
I've seen this on 2.5.11.

**Comment by johanandren** [comment]
Alright, verified with this minimal reproducer:

```
  Source.single(ByteString("whaa"))
    .runWith(FileIO.toPath(Paths.get("/i/just/made/it/up")))
    .onComplete(println)
```

Giving
```
Success(IOResult(0,Failure(java.nio.file.NoSuchFileException: /i/just/made/it/up)))
```

So I guess the fix for #23951 missed to fix this.
